sklearn/linear_model/_logistic.py

```diff
@@ -1798,6 +1814,11 @@ def fit(self, X, y, sample_weight=None):
         else:
             prefer = "processes"
 
+        # Remove the sample_weight parameter from score_params as it is
+        # explicitly being passed in the path_func.
+        score_params = routed_params.scorer.score
+        score_params.pop("sample_weight", None)
```

Comment:
those sample weights are used for _fit_, while the score params passed are used to score. The `_log_reg_scoring_path` with the current implementation doesn't take `sample_weight` into account for scoring since it doesn't pass it to the score method if `sample_weight` is passed. Either here it needs to not be popped, or there it needs to be handled explicitly.

Reply:
The reason I popped it is because sample_weight is also explicitly being passed to the _log_reg_scoring_path method. If we pass sample_weight directly and then also pass it through the score_params it just seems to be duplicated.